diff --git a/boost/boost.bzl b/boost/boost.bzl
index f733354..c6ed1da 100644
--- a/boost/boost.bzl
+++ b/boost/boost.bzl
@@ -154,7 +154,7 @@ def boost_deps():
         name = "bazel_skylib",
         sha256 = "1dde365491125a3db70731e25658dfdd3bc5dbdfd11b840b3e987ecf043c7ca0",
         urls = [
-            "https://mirror.bazel.build/github.com/bazelbuild/bazel-skylib/releases/download/0.9.0/bazel_skylib-0.9.0.tar.gz",
+            "https://qcraft-web.oss-cn-beijing.aliyuncs.com/cache/packages/bazel_skylib-0.9.0.tar.gz",
             "https://github.com/bazelbuild/bazel-skylib/releases/download/0.9.0/bazel_skylib-0.9.0.tar.gz",
         ],
     )
@@ -166,7 +166,7 @@ def boost_deps():
         sha256 = "c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1",
         strip_prefix = "zlib-1.2.11",
         urls = [
-            "https://mirror.bazel.build/zlib.net/zlib-1.2.11.tar.gz",
+            "https://qcraft-web.oss-cn-beijing.aliyuncs.com/cache/packages/zlib-1.2.11.tar.gz",
             "https://zlib.net/zlib-1.2.11.tar.gz",
         ],
     )
@@ -179,7 +179,10 @@ def boost_deps():
         build_file = "@com_github_nelhage_rules_boost//:BUILD.bzip2",
         sha256 = "ab5a03176ee106d3f0fa90e381da478ddae405918153cca248e682cd0c4a2269",
         strip_prefix = "bzip2-1.0.8",
-        url = "https://sourceware.org/pub/bzip2/bzip2-1.0.8.tar.gz",
+        urls = [
+            "https://qcraft-web.oss-cn-beijing.aliyuncs.com/cache/packages/bzip2-1.0.8.tar.gz",
+            "https://sourceware.org/pub/bzip2/bzip2-1.0.8.tar.gz",
+        ],
     )
 
     maybe(
@@ -189,6 +192,8 @@ def boost_deps():
         sha256 = "71928b357d0a09a12a4b4c5fafca8c31c19b0e7d3b8ebb19622e96f26dbf28cb",
         strip_prefix = "xz-5.2.3",
         urls = [
+            "https://qcraft-web.oss-cn-beijing.aliyuncs.com/cache/packages/xz-5.2.3.tar.gz",
+        ] + [
             "https://%s.dl.sourceforge.net/project/lzmautils/xz-5.2.3.tar.gz" % m
             for m in SOURCEFORGE_MIRRORS
         ],
@@ -201,7 +206,7 @@ def boost_deps():
         sha256 = "59ef70ebb757ffe74a7b3fe9c305e2ba3350021a918d168a046c6300aeea9315",
         strip_prefix = "zstd-1.4.4",
         urls = [
-            "https://mirror.bazel.build/github.com/facebook/zstd/releases/download/v1.4.4/zstd-1.4.4.tar.gz",
+            "https://qcraft-web.oss-cn-beijing.aliyuncs.com/cache/packages/zstd-1.4.4.tar.gz",
             "https://github.com/facebook/zstd/releases/download/v1.4.4/zstd-1.4.4.tar.gz",
         ],
     )
@@ -215,6 +220,7 @@ def boost_deps():
         sha256 = "7bd7ddceec1a1dfdcbdb3e609b60d01739c38390a5f956385a12f3122049f0ca",
         strip_prefix = "boost_1_76_0",
         urls = [
+            "https://qcraft-web.oss-cn-beijing.aliyuncs.com/cache/packages/boost_1_76_0.tar.gz",
             "https://boostorg.jfrog.io/artifactory/main/release/1.76.0/source/boost_1_76_0.tar.gz",
         ],
     )
@@ -225,5 +231,8 @@ def boost_deps():
         name = "openssl",
         sha256 = "17f5e63875d592ac8f596a6c3d579978a7bf943247c1f8cbc8051935ea42b3e5",
         strip_prefix = "boringssl-b3d98af9c80643b0a36d495693cc0e669181c0af",
-        url = "https://github.com/google/boringssl/archive/b3d98af9c80643b0a36d495693cc0e669181c0af.tar.gz",
+        urls = [
+            "https://qcraft-web.oss-cn-beijing.aliyuncs.com/cache/packages/boringssl-b3d98af9c80643b0a36d495693cc0e669181c0af.tar.gz",
+            "https://github.com/google/boringssl/archive/b3d98af9c80643b0a36d495693cc0e669181c0af.tar.gz",
+        ],
     )
