diff --git a/bazel/repository_locations.bzl b/bazel/repository_locations.bzl
index a95c5c5..bcb97c7 100644
--- a/bazel/repository_locations.bzl
+++ b/bazel/repository_locations.bzl
@@ -33,7 +33,7 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_desc = "Developer tools for working with Google's bazel buildtool.",
         project_url = "https://github.com/bazelbuild/buildtools",
         version = "4.2.4",
-        sha256 = "f3d86864a37663ad2303d8781e3f76e86b7e948cc6a3beae01da71cad2e302bd",
+        sha256 = "44a6e5acc007e197d45ac3326e7f993f0160af9a58e8777ca7701e00501c0857",
         release_date = "2021-12-03",
         strip_prefix = "buildtools-{version}",
         urls = ["https://github.com/bazelbuild/buildtools/archive/{version}.tar.gz"],
