diff --git a/fuzzing/init.bzl b/fuzzing/init.bzl
index 4c96927..da6405d 100644
--- a/fuzzing/init.bzl
+++ b/fuzzing/init.bzl
@@ -15,11 +15,9 @@
 """Dependency initialization utilities."""
 
 load("@rules_python//python:pip.bzl", "pip_install")
-load("@rules_python//python:repositories.bzl", "py_repositories")
 load("@bazel_skylib//:workspace.bzl", "bazel_skylib_workspace")
 
 def rules_fuzzing_init():
-    py_repositories()
     pip_install(
         name = "fuzzing_py_deps",
         extra_pip_args = ["--require-hashes"],
